core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -72,6 +72,26 @@ public static TestTable create(
     return new TestTable(ops, name);
   }
 
+  public static TestTable create(
```

Comment:
you might want to change the other `create` method to call this method and pass `ops` so that we don't duplicate the same implementations:

```
public static TestTable create(
      File temp,
      String name,
      Schema schema,
      PartitionSpec spec,
      SortOrder sortOrder,
      int formatVersion) {
    TestTableOperations ops = new TestTableOperations(name, temp);
    return create(temp, name, schema, spec, sortOrder, formatVersion, ops);
  }
```

Reply:
Indeed. Done.

Anyway, I was thinking of introducing a Builder for these test tables. I see that we have a number of variations now (with-without MetricsReporter, TableOperations and now in the partition stats PR we'll have one for props too) and might be cleaner to have a builder instead of these different create() methods. WDYT @nastra ?

Reply:
I'm fine if we start exploring and seeing how a Builder turns out with `TestTable` but I wouldn't do it as part of this PR

Reply:
Definitely not within this PR :)